src/metpy/plots/declarative.py

```diff
@@ -1262,3 +1266,248 @@ def _build(self):
+                    time_slice = ((self.data[dim_time] >= (self.time - self.time_window))
+                                  & (self.data[dim_time] <= (self.time + self.time_window)))
+                    data = self.data[time_slice].groupby(dim_stn).tail(1)
+                else:
+                    data = self.data.groupby(dim_stn).tail(1)
+                self._obsdata = data
+        return self._obsdata
+
+    @property
+    def plotdata(self):
+        """Return the data for plotting.
+
+        The data arrays, x coordinates, and y coordinates.
+
+        """
+        plot_data = {}
+        for dim_name in list(self.obsdata):
+            if dim_name.find('lat') != -1:
+                lat = self.obsdata[dim_name]
+            elif dim_name.find('lon') != -1:
+                lon = self.obsdata[dim_name]
+            else:
+                plot_data[dim_name] = self.obsdata[dim_name]
+        return lon.values, lat.values, plot_data
+
+    def draw(self):
+        """Draw the plot."""
+        if self._need_redraw:
+            if getattr(self, 'handle', None) is None:
+                self._build()
+            self._need_redraw = False
+
+    @observe('colors', 'symbol_mapper', 'locations', 'reduce_points')
+    def _set_need_rebuild(self, _):
+        """Handle changes to attributes that need to regenerate everything."""
+        # Because matplotlib doesn't let you just change these properties, we need
+        # to trigger a clear and re-call of contour()
+        self.clear()
+
+    def _build(self):
+        """Build the plot by calling needed plotting methods as necessary."""
+        lon, lat, data = self.plotdata
+
+        symbol_dict = {'current_weather': current_weather, 'high_clouds': high_clouds,
+                       'low_clouds': low_clouds, 'mid_clouds': mid_clouds,
+                       'pressure_tendency': pressure_tendency, 'sky_cover': sky_cover}
+
+        # Use the cartopy map projection to transform station locations to the map and
+        # then refine the number of stations plotted by setting a 300km radius
+        if self.parent._proj_obj == ccrs.PlateCarree():
+            scale = 1.
+        else:
+            scale = 100000.
+        point_locs = self.parent._proj_obj.transform_points(ccrs.PlateCarree(), lon, lat)
+        subset = reduce_point_density(point_locs, self.reduce_points * scale)
+
+        stationplot = StationPlot(self.parent.ax, lon[subset], lat[subset], clip_on=True,
+                                  transform=ccrs.PlateCarree(), fontsize=10)
+
+        for i, ob_type in enumerate(self.fields):
+            if len(self.locations) > 1:
+                location = self.locations[i]
+            else:
+                location = self.locations[0]
+            if len(self.colors) > 1:
+                color = self.colors[i]
+            else:
+                color = self.colors[0]
+            if self.formats[i] is not None:
+                if self.formats[i] in symbol_dict.keys():
```

Comment:
Could we replace `symbol_dict` with `getattr(wx_symbols, self.formats[i])`? We could either catch the `AttributeError` to trigger "no such mapper" or we could do:
```python
mapper = getattr(wx_symbols, self.formats[i], None)
if mapper is not None:
...
```

Reply:
I went with the second suggestion here to make it easier to fall back and try a text formatter.